src/main/java/io/github/bobocodebreskul/context/support/ReflectionUtils.java

```diff
@@ -1,16 +1,150 @@
 package io.github.bobocodebreskul.context.support;
 
 import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
 import lombok.experimental.UtilityClass;
 import lombok.extern.slf4j.Slf4j;
 
+/**
+ * Utility methods that are useful for reflective access.
+ */
 @Slf4j
 @UtilityClass
 public class ReflectionUtils {
 
-  public static boolean isAnnotationExistsFor(Class<? extends Annotation> annotation, Class<?> clazz) {
-    log.trace("Scanning class {} for @{} existence", clazz.getName(), annotation.getSimpleName());
+  /**
+   * Checks if a specified annotation is present on the given class.
+   *
+   * @param annotation The annotation type to check for. Must not be null.
+   * @param clazz      The class on which to check for the presence of the annotation. Must not be
+   *                   null.
+   * @return {@code true} if the specified annotation is present on the given class, {@code false}
+   * otherwise.
+   * @throws NullPointerException If either the {@code annotation} or {@code clazz} parameter is
+   *                              {@code null}.
+   * @apiNote This method performs a simple check for the presence of the specified annotation on
+   * the class. It does not support recursive search for annotations on superclasses or interfaces.
+   * The presence of the annotation is determined using the {@link Class#isAnnotationPresent(Class)}
+   * method.
+   */
+  public static boolean isAnnotationPresentForClass(Class<? extends Annotation> annotation,
+      Class<?> clazz) {
+    Objects.requireNonNull(annotation, "The annotation parameter cannot be null!");
+    Objects.requireNonNull(clazz, "The class parameter cannot be null!");
+    log.trace("Scanning class {} for@ @{} existence", clazz.getName(), annotation.getSimpleName());
     return clazz.isAnnotationPresent(annotation);
   }
 
+
+  /**
+   * Checks if a specified annotation is present on the given constructor.
+   *
+   * @param annotation  The annotation type to check for. Must not be null.
+   * @param constructor The constructor on which to check for the presence of the annotation. Must
+   *                    not be null.
+   * @return {@code true} if the specified annotation is present on the given constructor,
+   * {@code false} otherwise.
+   * @throws NullPointerException If either the {@code annotation} or {@code constructor} parameter
+   *                              is {@code null}.
+   */
+  public static boolean isAnnotationPresentForConstructor(Class<? extends Annotation> annotation,
```

Comment:
it can be private

Reply:
We can make it package private, as it's already covered with tests.